### PR TITLE
Fix unexpected behavior of symmetry_conserving_bravyi_kitaev

### PR DESCRIPTION
**Pull request by r-imai-quantum**
This pull request fixes the unexpected behavior of symmetry_conserving_bravyi_kitaev.

It is the following:
```python
>>> symmetry_conserving_bravyi_kitaev(FermionOperator("3^ 3"), 8, 4)
0.5 [] +
-0.5 [Z0 Z1]
```
This is caused by prune_unused_indices, which deletes all unused indices.

The expected result is
```python
0.5 [] +
-0.5 [Z3 Z4]
```

### CODE REVIEW

**Comment by r-imai-quantum** [comment]
Could someone please review it?

**Comment by r-imai-quantum** [comment]
Most of the SCBK features seem to be written by @sammcardle30 and @babbush.
Would it be possible for you to review changes in this pull request?

File: src/openfermion/transforms/opconversions/remove_symmetry_qubits.py [hunk 1]
```diff
@@ -131,3 +133,23 @@ def edit_hamiltonian_for_spin(qubit_hamiltonian, spin_orbital, orbital_parity):
     qubit_hamiltonian.compress()
 
     return qubit_hamiltonian
+
+
+def remove_indices(symbolic_operator, indices):
```
Comment (ncrubin) [inline src/openfermion/transforms/opconversions/remove_symmetry_qubits.py:138]: Just a small nit could you add a more detailed description of what this function is doing and type annotations?  A minimal requirement is a description of input parameters and types.  even for a small function like this.

**Review by ncrubin — CHANGES_REQUESTED** [review]

File: src/openfermion/transforms/opconversions/remove_symmetry_qubits.py (hunk 1 above)
Comment (r-imai-quantum) [reply src/openfermion/transforms/opconversions/remove_symmetry_qubits.py:138]: Thank you for reviewing. Linting errors has been fixed.

**Review by ncrubin — APPROVED** [review]

**Comment by r-imai-quantum** [comment]
Thank you!

**Pull request merged**